src/backend/marsha/settings.py

```diff
@@ -417,54 +435,24 @@ class Production(Base):
 
     ALLOWED_HOSTS = values.ListValue(None)
 
-    # For static files in production, we want to use a backend that includes a hash in
-    # the filename, that is calculated from the file content, so that browsers always
-    # get the updated version of each file.
     STATICFILES_STORAGE = values.Value(
-        "marsha.core.storage.ConfigurableManifestS3Boto3Storage"
+        "whitenoise.storage.CompressedManifestStaticFilesStorage"
```

Comment:
Can you confirm that this takes care of invalidating cdn cache when the file content changes?

Reply:
For what I understand, this is managed with the response headers : http://whitenoise.evans.io/en/stable/base.html#caching-headers

Reply:
Actualy headers are information for the CDN but that's not what we want. From this setting's description http://whitenoise.evans.io/en/stable/django.html#WHITENOISE_KEEP_ONLY_HASHED_FILES I understand that hashed file names are generated during collectstatic and that you should be careful not to keep the original files to optimize the size of your Docker image.

Reply:
Invalidation is automatically taken care of, with versioned files (through hashes) kept forever http://whitenoise.evans.io/en/stable/django.html#WHITENOISE_MAX_AGE

Reply:
On my other project with WhiteNoise I don't need the `WHITENOISE_KEEP_ONLY_HASHED_FILES` to have hashed files in the image.

OTOH I don't have many statics there so the `collectstatic` is nearly instantaneous

Reply:
Yes indeed all fiels are hashes and unique so there is no cache issue. I will set the `WHITENOISE_KEEP_ONLY_HASHED_FILES` setting.

Reply:
Done

Reply:
I removed the `WHITENOISE_KEEP_ONLY_HASHED_FILES` setting because it removes the files generated by webpack and then the needed files are missing in the image.

Reply:
Ok! Then you can use rdfind to optimize while keeping all the files: https://github.com/openfun/openedx-docker/blob/master/releases/master/0/bare/Dockerfile#L139

Reply:
@sampaccoud Done.